src/extras/primitives/primitives.js

```diff
@@ -30,6 +31,29 @@ module.exports.registerPrimitive = function registerPrimitive (name, definition)
       createdCallback: {
         value: function () {
           if (definition.deprecated) { console.warn(definition.deprecated); }
+          this.resolveMappingCollisions();
+        }
+      },
+
+      /**
+       * If a mapping collides with a registered component name
+       * it renames the mapping to componentname-property
+       */
+      resolveMappingCollisions: {
+        value: function () {
+          var mappings = this.mappings;
+          var self = this;
+          Object.keys(mappings).forEach(function resolveCollision (key) {
+            var newAttribute;
+            if (components[key]) {
+              newAttribute = mappings[key].replace('.', '-');
+              mappings[newAttribute] = mappings[key];
```

Comment:
Alternative API might be: `<a-box style="width: 5; color: red">`, containing the mappings within a reserved component?